visualdl/storage/tablet.cc

```diff
@@ -13,9 +13,25 @@ See the License for the specific language governing permissions and
 limitations under the License. */
 
 #include "visualdl/storage/tablet.h"
+#include "visualdl/storage/storage.h"
 
 namespace visualdl {
 
+void Tablet::SetTag(const std::string& mode, const std::string& tag) {
+    auto internal_tag = mode + "/" + tag;
+    string::TagEncode(internal_tag);
+    internal_encoded_tag_ = internal_tag;
+    data_->set_tag(internal_tag);
+    WRITE_GUARD
+}
+
+Record Tablet::AddRecord() {
+    parent()->MarkTabletModified(internal_encoded_tag_);
```

Comment:
MarkTableModified use a `set` and that takes an overhead if `AddRecord` is triggered frequently(e.g. in a scalar), and theoretically, it should be placed in `WRITE_GUARD` so that modifying a record will also mark the tablet "modified". But currently, this implementation is ok.

There is a trick that can be faster,  assigning each tablet an internal member `id_` (continuously ascending from zero), and make `modified_tablet_set_` a `vector`, the marking logic like this

```c++
vector<bool> modified_tablet_set_; // each time add a tablet will append a true
Record Tablet::AddRecord() {
  parent()->MarkTabletModified(id_);
  // ...
}
inline void Storage::MarkTabletModified(int tagid) {
  modified_tablet_set_[tagid] = true;
}
void Storage::PersistToDisk(const std::string& dir) ... // clear states
```